Reinstate old RPC API methods

pnax opened the pull request:
## Purpose

Remove deprecation warning and comments to all old RPC API methods and mark the new methods as experimental.
What were the deprecated methods (and old stable) are again the default stable API methods.

## Context

https://github.com/zonemaster/zonemaster/issues/1159

Relates to https://github.com/zonemaster/zonemaster-backend/pull/1054 and https://github.com/zonemaster/zonemaster-backend/pull/1083.

## Changes

Revert commits where old methods were deprecated and based upon new methods. Now the experimental methods are based on the stable one.

## How to test this PR

* Unit tests should pass
* Review the code and verify that the unit tests cover the stable API
* Check that the new experimental API is working (there is currently no unit test)
* Check that changes made in Backend for this release are still working (and have not been inadvertently reverted)

mattias-p reviewed (dismissed):
I haven't tested but this looks good to me.

mattias-p commented:
In the psgi file there's a check for `$ini->RPCAPI_enable_user_create or $ini->RPCAPI_enable_add_batch_user`. Since RPCAPI.enable_add_batch_user is truthy by default, the only way to disable the methods is to disable RPCAPI.enable_add_batch_user (and leave RPCAPI.enable_user_create disabled), right?

Also from my point of view it would be better to forbid specifying both RPCAPI.enable_add_batch_user and RPCAPI.enable_user_create in the same config.

pnax commented:
I've updated the PR to be aligned with the current documentation in https://github.com/zonemaster/zonemaster/pull/1162. And as suggested, an error is now raised if both stable and experimental parameters are defined in the configuration file.

mattias-p commented on t/test01.t:
```diff
@@ -254,35 +251,32 @@ subtest 'domain_history' => sub {
-        $domain_history = $res_domain_history->{history};
-        is( scalar( @$domain_history ), 2, 'Only 2 tests should be retrieved' );
+        $test_history = $rpcapi->get_test_history( $method_params );
+        #diag explain( $test_history );
```
Just a thought: Would it make sense to call `note` here instead of having a commented out call to `diag`?

pnax replied:
Good suggestion. I've updated.

mattias-p reviewed (approved):
LGTM!

pnax commented:
To quote comment from the documentation PR (https://github.com/zonemaster/zonemaster/pull/1162#issuecomment-1587474136):

AFAICT this PR is done. I've gotten a few reviews, addressed all comments up to this point and everybody seems ok with the implementation.

> I'm leaving it open until the implementation is done too so they can be merged at the same time.

I think we reached that point.

matsduf commented:
I think this can be merged.
